File: src/main/java/com/github/cybellereaper/wizpets/api/talent/TalentRegistryView.java
package com.github.cybellereaper.wizpets.api.talent;

/**
 * Read-only view of the registered talent factories.
 */
public interface TalentRegistryView extends Iterable<PetTalentDescriptor> {
    PetTalentDescriptor get(String id);

    int size();
}

File: src/main/kotlin/com/github/cybellereaper/wizpets/api/talent/TalentRegistryView.kt
package com.github.cybellereaper.wizpets.api.talent

/**
 * Read-only view of the registered talent factories.
 */
interface TalentRegistryView : Iterable<PetTalentDescriptor> {
    operator fun get(id: String): PetTalentDescriptor?
    val size: Int
}

data class PetTalentDescriptor(
    val id: String,
    val displayName: String,
    val description: String
)
